### DISCUSSION TITLE
How to point UWP Apps to a specific version of webview2 runtime that was tested and published to the store

### DISCUSSION THREAD

**Discussion by DotNet-Fan** [MicrosoftEdge/WebView2Feedback#4271] [Q&A] [answered]
This is a question with reference to 
https://learn.microsoft.com/en-us/answers/questions/1471087/how-to-point-to-a-specific-version-of-webview2-run

In UWP app , we reference the Microsoft.UI.Xaml nuget package and uses the webview2 from Microsoft.UI.Xaml.Controls namespace.
When we published the app to the store , the webview2 runtime was 114.***.*** and some of the major functioanlities were working. Now the latest webview2 runtime updates has broken the functionalities and it has affected the App's usage. 
What happens is whenever the WebView2 runtime is updated thorough the windows updates , we see webview2 automatically is pointing the latest ones in the taskmanager.
And any webview2 fixes are taking more time to get it released publicly.
How can we make sure that UWP app points to the webview2 runtime that was tested and published to the store?

Thanks

**Comment by victorhuangwq** [comment]
The instructions of how to select a webview2 fixed runtime is as follows: 
https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/distribution#details-about-the-fixed-version-runtime-distribution-mode

**Comment by DotNet-Fan** [comment]
Thanks @victorhuangwq  .  
In the documentation that you have shared , it is mentioned like this
```
<ItemGroup>
   <Content Include="FixedRuntime\(version number)\\**\*.*">
   <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
   </Content>
</ItemGroup>
```
So the UWP app supports x86 and x64 and how do we include webview2 x64 runtime libs and WebView2 x86 libs based on the build and publish configuration?

**Comment by DotNet-Fan** [comment]
@victorhuangwq  i have managed to build and copy the fixedruntime  119.0.2151.97 based on the build configuration. 
And i set the environment variable to the path of the webview2 runtime 119.0.2151.97 .
Environment.SetEnvironmentVariable("WEBVIEW2_BROWSER_EXECUTABLE_FOLDER", fixedPath);

But i still see in the taskmanager that it is picking from the latest webview2 runtime 120.x.xxxx.xx. Any ideas why it is picking the latest and not the fixed runtime path from the environment variable?

**Reply by victorhuangwq** [reply]
Did you do the following?
> Verify that the bin\**designated architecture**\Release folder has a matching FixedRuntime\(version number) folder with the runtime files in it.

**Comment by DotNet-Fan** [comment, marked as answer]
@victorhuangwq One webview instance was picking the default path and another from Environment variable. So the instance that picks the initial path  first would be the path for all the other webview instances.

**Reply by victorhuangwq** [reply]
Does this mean your issue has been resolved? Or are you still looking for help

**Comment by contactteckee** [comment]
@victorhuangwq Yes this is fixed and Thanks.
